app/src/main/java/com/kickstarter/ui/views/ReferrerBreakdownView.java

```diff
@@ -0,0 +1,74 @@
+package com.kickstarter.ui.views;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.RectF;
+import android.util.AttributeSet;
+import android.view.View;
+
+import com.kickstarter.libs.ReferrerColor;
+
+public class ReferrerBreakdownView extends View {
+
+  public Canvas canvas;
+  private Double customSweepAngle;
+  private Double externalSweepAngle;
+  private Double internalSweepAngle;
+  private RectF outerRectangle;
+  private RectF innerRectangle;
+  private Paint paint;
+
+  public ReferrerBreakdownView(final Context context, final AttributeSet attributeSet) {
+    super(context, attributeSet);
+    this.paint = new Paint();
+    this.outerRectangle = new RectF(0, 0 , 0, 0);
+    this.innerRectangle = new RectF(0, 0 , 0, 0);
```

Comment:
doing this in the intializer because our linter helpfully informed me that `onDraw` can get called lots and lots of times it's bad to allocate things there.